### PR TITLE
server: Simplify passing logger setup by passing only logger

### PR DESCRIPTION
**Pull request by serathius**
Follow up from https://github.com/etcd-io/etcd/issues/12326
cc @ptabor

### CODE REVIEW

File: server/embed/etcd.go [hunk 1]
```diff
@@ -211,9 +211,6 @@ func StartEtcd(inCfg *Config) (e *Etcd, err error) {
 		CorruptCheckTime:                         cfg.ExperimentalCorruptCheckTime,
 		PreVote:                                  cfg.PreVote,
 		Logger:                                   cfg.logger,
-		LoggerConfig:                             cfg.loggerConfig,
```
Comment (ptabor) [inline server/embed/etcd.go:214]: Shell these properties (cfg.loggerConfig/Core/WriteSyncer) be removes as well, as they become a `no op` that can be misleading.

Comment (serathius) [reply server/embed/etcd.go:214]: Done.

**Review by ptabor — APPROVED** [review]

**Review by lilic — COMMENTED** [review]
Nice!

Curious should we mention this in the release notes, as its an exposed server configuration, or what do we consider as public API breaking changes?

**Comment by ptabor** [comment]
I think that exported properties of the 'embed' package has not changed, so its an `internal` change.

**Comment by gyuho** [comment]
> I think that exported properties of the 'embed' package has not changed, so its an internal change.

If we are going to ignore the logging fields from `embed.Config`, can we highlight these in CHANGELOG and maybe remove those fields in 3.6?

**Comment by ptabor** [comment]
embed.Config never accepted `loggerCore` & `loggerConfig` publicly: 

https://github.com/etcd-io/etcd/blob/ddc4f473c98d07c6a812b4d0f844e19c7216baf2/server/embed/config.go#L381-L383

The way to configure it was to pass: 
https://github.com/etcd-io/etcd/blob/ddc4f473c98d07c6a812b4d0f844e19c7216baf2/server/embed/config.go#L368

So the breaking change already happened in: eafbc8c57efc716644c328f63677ca7eadeebdfe where the signature got simplified. 
Will provide small PR that brings back the compatiblity.

**Comment by ptabor** [comment]
Brought back the original API (and changelog in #12973)

**Pull request merged**